### PR TITLE
Quick fix

### PR DESCRIPTION
**Pull request by freddieknets**
Quick fix: in `UnionRef._to_json()` the returned class name should be that of the underlying type.

Without this fix, the returned class name will be that of the `UnionRef` itself, failing to instantiate a new `UnionRef` from this json ("`TypeError: Uref is not member of <unionref Uref>`").

This passed unnoticed as it works correctly for an array of `UnionRef`, hence I added a test for the bare `UnionRef` as well to catch this in the future.

As a side remark: it is possible to instantiate an array of `UnionRef` directly from a list of `tuple`, though this is not possible for a bare `UnionRef` where the `tuple` has to be unpacked. Is this intentional? Similarly, the underlying `XoStruct` fields can be directly accessed in case of an array of `UnionRef`, while one has to call `.get()` on the bare `UnionRef`.

The difference between these two can be clearly seen in the two tests inside `test_to_json.py`. This is not an issue, though it might be interesting to homogenise the API.

### CODE REVIEW

**Comment by rdemaria** [comment]
better 
`self.get().__class__.__name__`  -> `v.__class__.__name__`
because self.get() is a little expensive

The unpacking is perhaps a bug,

https://github.com/rdemaria/xobjects/blob/ffa8c7b0f6c1a4c72c4448db1a0f28adc295f8e5/xobjects/ref.py#L242

it could be `args` -> `args[0]`  or there is a reason I cannot remember at the moment.

**Comment by freddieknets** [comment]
Replaced the `self.get().__class__.__name__` with `v.__class__.__name__` as suggested, and also assigned the `._to_json()` in the `if` above to a new variable `w` (because otherwise, in case the field has indeed a `._to_json()` attribute itself, the variable `v` would now be a `dict` and the returned class name would be wrong).

File: xobjects/ref.py [hunk 1]
```diff
@@ -297,9 +297,10 @@ def _get_inner_types(cls):
 
     def _to_json(self):
         v = self.get()
+        w = v
```
Comment (rdemaria) [inline xobjects/ref.py]: what about `classname=v.__class__.__name__`?  instead of `w=v`  which is a bit obscure

**Comment by freddieknets** [comment]
Agreed, and implemented.

**Pull request merged**